### PR TITLE
Prevent exception in the log

### LINKED ISSUE #22200: External Link check for documentation logs warning and exception: WARN FilenoUtil: Native subprocess control requires open access
### Description

The build logs a warning: 

https://github.com/keycloak/keycloak/actions/runs/5739004142/job/15553733564?pr=22187

```
2023-08-02T12:50:04.748Z [main] WARN FilenoUtil : Native subprocess control requires open access to sun.nio.ch
Pass '--add-opens java.base/sun.nio.ch=org.jruby.dist' or '=org.jruby.core' to enable.
java.lang.IllegalCallerException: sun.nio.ch is not open to unnamed module @5b275dab
	at java.base/java.lang.Module.addOpens(Module.java:836)
	at com.headius.backport9.modules.impl.Module9.addOpens(Module9.java:28)
	at com.headius.backport9.modules.Modules.addOpens(Modules.java:22)
	at org.jruby.util.io.FilenoUtil$ReflectiveAccess.<clinit>(FilenoUtil.java:140)
	at org.jruby.util.io.FilenoUtil.getFilenoUsingReflection(FilenoUtil.java:111)
	at org.jruby.util.io.FilenoUtil.filenoFrom(FilenoUtil.java:107)
	at org.jruby.util.io.ChannelFD.initFileno(ChannelFD.java:42)
	at org.jruby.util.io.ChannelFD.<init>(ChannelFD.java:32)
	at org.jruby.util.io.OpenFile.setChannel(OpenFile.java:196)
	at org.jruby.RubyIO.prepIO(RubyIO.java:261)
	at org.jruby.RubyIO.prepStdio(RubyIO.java:190)
	at org.jruby.RubyGlobal.initSTDIO(RubyGlobal.java:297)
	at org.jruby.RubyGlobal.createGlobals(RubyGlobal.java:211)
	at org.jruby.Ruby.init(Ruby.java:1260)
	at org.jruby.Ruby.newInstance(Ruby.java:370)
	at org.jruby.javasupport.JavaEmbedUtils.initialize(JavaEmbedUtils.java:79)
	at org.asciidoctor.jruby.internal.JRubyAsciidoctor.createRubyRuntime(JRubyAsciidoctor.java:132)
	at org.asciidoctor.jruby.internal.JRubyAsciidoctor.<init>(JRubyAsciidoctor.java:52)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:789)
	at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:729)
	at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1403)
	at org.asciidoctor.Asciidoctor$Factory.create(Asciidoctor.java:485)
	at org.keycloak.documentation.test.Config.loadDocumentAttributes(Config.java:138)
	at org.keycloak.documentation.test.Config.<init>(Config.java:54)
	at org.keycloak.documentation.test.Config.<clinit>(Config.java:21)
	at org.keycloak.documentation.test.Guide.<init>(Guide.java:29)
	at org.keycloak.documentation.test.ExternalLinksTest.checkExternalLinks(ExternalLinksTest.java:16)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
```

### Discussion

_No response_

### Motivation

There shouldn't be unrelated warnings and stack traces in the build log

### Details

In addition to that, it tests seems to test too many external links, and they are not cached for local runs

### PR DESCRIPTION
**Pull request by ahus1**
Also speed up the external link check by avoiding checking each bug submission link in the rendered docs which only differs by its parameter.

The external link check no completes in 2 minutes instead of 9 minutes on GitHub actions.

Closes #22200

### CODE REVIEW

**Comment by ahus1** [comment]
@miquelsi - as you've reviewed PR #22187, I wonder if you'd feel confident to review this PR as well. 

It avoids an exception in the log and speeds up the build by a few minutes. 

As #22187 hasn't been merged yet, it shows the error in the external links (as it should). 

Thanks!

**Comment by ahus1** [comment]
@mhajas - I wonder if you could have a review for this one when you have the time. Thanks!

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -85,18 +79,12 @@ jobs:
...
         run: |
           MVN_HTTP_CONFIG="-Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.http.retryHandler.class=standard -Dmaven.wagon.http.retryHandler.count=3 -Dmaven.wagon.httpconnectionManager.ttlSeconds=120"
-          ./mvnw install -DskipTests -am -pl docs/documentation/tests -nsu -B -e -Pdocumentation $MVN_HTTP_CONFIG 
-
-      - id: test-documentation
-        name: Verify Keycloak documentation
-        shell: bash
-        run: |
-          ./mvnw test -Dtest=ExternalLinksTest -am -pl docs/documentation/tests,docs/documentation/dist -nsu -B -e -Pdocumentation
+          ./mvnw install -Dtest=ExternalLinksTest -am -pl docs/documentation/tests -nsu -B -e -Pdocumentation $MVN_HTTP_CONFIG
```
Comment (mhajas) [inline .github/workflows/documentation.yml:87]: Should we add `docs/documentation/dist` to running tests similarly as it was before? 
```suggestion
          ./mvnw install -Dtest=ExternalLinksTest -am -pl docs/documentation/tests,docs/documentation/dist -nsu -B -e -Pdocumentation $MVN_HTTP_CONFIG 
```

**Review by mhajas — COMMENTED** [review]
Thank you for the PR @ahus1, I added one comment.

File: .github/workflows/documentation.yml (hunk 1 above)
Comment (ahus1) [reply .github/workflows/documentation.yml:87]: @mhajas - I had a look and gave it another try: To run the tests, it needs only the dependencies `docs/documentation/tests` which are the different rendered HTML outputs, not the `docs/documentation/dist`, therefore this additional project is not needed here. 

The `dist` part is only needed in the regular build which is triggered in the `build` which then also uploads these files as a build result.

Comment (mhajas) [reply .github/workflows/documentation.yml:87]: Thanks for looking into this

**Review by mhajas — APPROVED** [review]
Thanks @ahus1

**Pull request merged**